test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/Profiler/ProfilerServiceTests.cs

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlServer.Management.XEvent;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Admin;
+using Microsoft.SqlTools.ServiceLayer.Admin.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.DisasterRecovery;
+using Microsoft.SqlTools.ServiceLayer.DisasterRecovery.Contracts;
+using Microsoft.SqlTools.ServiceLayer.FileBrowser;
+using Microsoft.SqlTools.ServiceLayer.FileBrowser.Contracts;
+using Microsoft.SqlTools.ServiceLayer.IntegrationTests.Utility;
+using Microsoft.SqlTools.ServiceLayer.Profiler;
+using Microsoft.SqlTools.ServiceLayer.Profiler.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Test.Common;
+using Moq;
+using Xunit;
+
+namespace Microsoft.SqlTools.ServiceLayer.IntegrationTests.Profiler
+{
+    public class ProfilerServiceTests
+    {
+        /// <summary>
+        /// Verify that a start profiling request starts a profiling session
+        /// </summary>
+        [Fact]
+        public async Task TestHandleStartAndStopProfilingRequests()
+        {
+            // initialize test state
+            ProfilerService profilerService = new ProfilerService();
+            var liveConnection = LiveConnectionHelper.InitLiveConnectionInfo("master");
```

Comment:
in my test I use  using (SelfCleaningTempFile queryTempFile = new SelfCleaningTempFile()) to create unique uri,  otherwise other tests might use the uri you are using and it can cause the connection to fail for your  test randomly

Reply:
@llali thanks!  I'll use that instead.